spec.md

```diff
@@ -91,13 +91,37 @@ perform a scheduled reboot.
 #### Event
 
 An "event" is a data record expressing an occurrence and its context. Events are
-routed from an event producer (the source) to interested event consumers. The
-routing can be performed based on information contained in the event, but an
-event will not identify a specific routing destination. Events will contain two
-types of information: the [Data](#data) representing the Occurrence and
+routed from an event producer to interested event consumers. The routing can be
+performed based on information contained in the event, but an event will not
+identify a specific routing destination. Events will contain two types of
+information: the [Data](#data) representing the Occurrence and
 [Context](#context) metadata providing contextual information about the
 Occurrence.
 
+#### Producer
+
+The "producer" is a specific instance, process or device that creates the data
+structure describing the CloudEvent.
+
+#### Source
+
+The "source" is the logical system or service where the occurrence happened. In
```

Comment:
While I will accept the will of the group to accept this PR, this appears to contradict the definition of the `source` attribute as [defined](https://github.com/cloudevents/spec/blob/master/spec.md#source):

```
Description: This describes the event producer.
```

Given this, it seems that either the two should be disambiguated or this PR should be updated to change that text before it is merged.

Reply:
Thanks for bringing this up. I was aware of this contradiction and wanted to stir up the discussion, because "describes the event producer" seems to me very vague. Why would we call a field, that describes the producer, _source_ and not _producer_? 
I think the fact the source attribute is a URI (an identifier and not descriptor) and the examples like `urn:event:from:myapi/resource` suggest that the source attribute identifies something more logical. I really have a hard time finding better words. Any proposals for a better wording are welcome.

Reply:
I'm not sure I have terminology but perhaps its useful to consider that perhaps the "logical" source exists in user/application conceptual design space while producers live more in the space of system implementation?  That seems the strongest argument for the #420 definition of source.

Perhaps a downside is that making it logical reduces the attribute's value to middleware or consumers.  Personally, I had leaned towards preferring the source as a unique producer (guaranteed by application), primarily because of the relationship (assuming ordering guarantees) that ensures a map of producers to lists of ordered events which looks much like a logical clock if not a CRDT.  If the IDs of producers are monotonically increasing and the event IDs are monotonically increasing, one also gets efficient and simple deduplication+idempotency which in my favored scenarios of many multi-versioned consumer groups makes the costs tractable.  I think I, at least (if not fully), failed in my reading of the definition of source because the understanding I retained from some time ago, by bias, weighted that first sentence strongly and broadly ignored the implications of the remainder.

Given the approval of this PR then it seems that solution is to simply remove mention of producers from the definition of `source`.